tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,272 @@
+            'zipfile54':'https://nihcc.box.com/shared/static/z7s2zzdtxe696rlo16cqf5pxahpl8dup.zip',
+            'zipfile55':'https://nihcc.box.com/shared/static/shr998yp51gf2y5jj7jqxz2ht8lcbril.zip',
+            'zipfile56':'https://nihcc.box.com/shared/static/kqg4peb9j53ljhrxe3l3zrj4ac6xogif.zip',
+            'ann_file':'https://raw.githubusercontent.com/anir16293/Deep-Lesion/master/DL_info.csv'
+    }
+    dl_manager._register_checksums = True
+    paths = dl_manager.download_and_extract(resources)
+    ann_path = paths['ann_file']
+    del paths['ann_file']
+    lut = _build_lut(paths)
+    train_split, val_split, test_split = Split(ann_path).split_generator()
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=100,
+            gen_kwargs={
+                "lut": lut,
+                "split": train_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=10,
+            gen_kwargs={
+                "lut": lut,
+                "split": val_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=10,
+            gen_kwargs={
+                "lut": lut,
+                "split": test_split,
+            },
+        ),
+    ]
+
+
+  def _generate_examples(self, lut, split):
+    """Returns examples
+    :type lut: dict, lookup dictionary to lookup whole paths of series_folder
+    :type split: string, path to read ann files
+    :rtype: yield idx, example
+    """
+    for idx, value in enumerate(split.values):
+      fn, bboxs, size, sp = value
+      record = {
+          "image/name": fn,
+          "image": _lookup_image_path(lut, fn),
+          "bbox": _format_bboxs(bboxs, size)
+      }
+      yield idx, record
+
+    
+def _build_lut(paths):
+    """Returns lookup dictionary to lookup whole paths of series_folder
+    :type paths: dict, {<zipfn>:<unzip_path>}
+    :rtype: dict, {<series_folder>:<unzip_path>/Images_png/<series_folder>}
+    """
+    lut = {}
+    for k, v in paths.items():
+        for fd in tf.io.gfile.listdir(os.path.join(v,'Images_png')):
+            lut[fd] = os.path.join(v,'Images_png', fd)
+    return lut
+        
+
+def _lookup_image_path(lut, fn):
+    """Returns whole path of an image
+    """
+    slice_fn = fn.split('_')[-1]
+    fd = '_'.join(fn.split('_')[:-1])
+    
+    return os.path.join(lut[fd], slice_fn)
+    
+    
+def _format_bboxs(bboxs, size):
+    """Return bbox feature
+    :type bboxs: string, "xxx,xxx,xxx,xxx"
+    :type size: string, "xxx,xxx"
+    :rtype: tfds.features.BBox
+    """
+    size = list(map(lambda x: float(x), size.split(',')))
+    coords = list(map(lambda x: float(x), bboxs.split(',')))
+    coords = list(map(lambda x: x if x>0 else 0.0, coords))
+    coords = list(map(lambda x: x if x<size[0] else size[0], coords))
```

Comment:
If I understood this code correctly, you can replace it with:

coords = np.clip([float(x) for x in bbox.split(',')], 0.0, size[0])

which brings the question whether this is correct approach?
is size[0] == size[1]? if not - shouldn't you clip each dimension separately?

The alternative is to clip the xmin, ymin, xmax, ymax to 0.0, 1.0 after the division.

Reply:
> If I understood this code correctly, you can replace it with:
> 
> coords = np.clip([float(x) for x in bbox.split(',')], 0.0, size[0])
> 
> which brings the question whether this is correct approach?
> is size[0] == size[1]? if not - shouldn't you clip each dimension separately?
> 
> The alternative is to clip the xmin, ymin, xmax, ymax to 0.0, 1.0 after the division.

Thanks for the advice. I updated it. the sizes of two dimensions are always the same for this dataset. so it should works here.